pkg/api/v1beta1/types.go

```diff
@@ -397,6 +397,23 @@ type EnvVar struct {
 	Key  string `json:"key,omitempty" description:"name of the environment variable; must be a C_IDENTIFIER; deprecated - use name instead"`
 	// Optional: defaults to "".
 	Value string `json:"value,omitempty" description:"value of the environment variable; defaults to empty string"`
+	// Optional: specify a source for the value of the EnvVar.
+	Source *EnvVarSource `json:"source,omitempty" description:"source for the environment variable's value; cannot be used if value is not empty"`
```

Comment:
I feel like this could be named more obviously, still.

ValueFrom?  Eval?

I'm just thinking about how the JSON reads - what's the story that it tells?  "Name 'foo', value from pod field v1beta1.ID" is a pretty good story, better than ""Name 'foo', source pod field v1beta1.ID".  and it ValueFrom makes the mutual exclusion more obvious.